watcher/internal/mocks/doc.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2021 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package mocks contains generated mocks for use in tests. These mocks are
+// generated using testify mock framework.
+//
+// Command to generate the mocks are defined as "go generate" directives in the
+// file where the interfaces are defined. However, some mocks might be manually
```

Comment:
Regarding the file name: I think it is not required. Because, the documentation of the generated mock will contain the name of the interface that it is mocking.

Regarding manual adaptation of mocks. I was re-thinking if it is possible to avoid this and figured out a solution. In this PR, i have updated the docs, to remove the phrases that say the mocks will be modified.

Actual fix, i have included in #216. Previously, the mocks were modified in #216 (5/5). Now, the i have updated the PR to not modify the mocks and use an alternate approach instead.